### DISCUSSION TITLE
How to enable QuartzHealthChecks?

### DISCUSSION THREAD

**Discussion by MarkLFT** [quartznet/quartznet#2549] [Q&A]
My project uses Quartz.AspNetCore version 3.113.  I read in several posts regarding QuartzhealthChecks being available in QuartzAspNetCore, but the only command I can find to enable them is:
```
.AddTypeActivatedCheck<Quartz.AspNetCore.HealthChecks.QuartzHealthCheck>("quartz-scheduler")
```
But when I use this, it says QuartzHealthCheck is not accessible because of its protection level.   But without this line, if I check the /heath API, I see my other health checks, SQL Server, RabbitMQ, and Internet, but nothing for Quartz.

How do I enable health checks for Quartz so I can see the status?

**Comment by lahma** [comment]
Currently if you use [the ASP.NET Core integration package](https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/aspnet-core-integration.html), Quartz will add the health check for you if you are on NET 6 or later:

https://github.com/quartznet/quartznet/blob/b30443eb5a8a95da2c706e45323a045909f9859e/src/Quartz.AspNetCore/AspNetCore/QuartzServiceCollectionExtensions.cs#L13-L24

**Comment by MarkLFT** [comment]
Ok, I was using AddQuartzHostedService, not realising that AddQuartzServer call AddQuartzHostedServer.  I think it would be helpful if that was mentioned in the documentation, as for me I spent hours trying to figure out how to achieve this, and I was simply using the wrong method call.

**Reply by lahma** [reply]
Please send a PR to improve the documentation 👍🏻

**Reply by MarkLFT** [reply]
Trying to get a project out the door at the moment, but will happily propose some changes as soon as I have a moment.
